modules/tinymce/src/core/test/ts/browser/delete/BeforeInputTest.ts

```diff
@@ -0,0 +1,56 @@
+import { Keys } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyContentActions, TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.delete.BeforeInputTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce'
+  }, [], true);
+
+  const testBackspace = (
+    setupHtml: string,
+    setupPath: number[],
+    setupOffset: number,
+    expectedHtml: string,
+    cancelBeforeInput: boolean
+  ) => {
+    const editor = hook.editor();
+    const inputEvents: string[] = [];
+    const collect = (event: InputEvent) => {
+      inputEvents.push(event.inputType);
+    };
+    const beforeInputCollect = (event: InputEvent) => {
+      collect(event);
+
+      if (cancelBeforeInput) {
+        event.preventDefault();
+      }
+    };
+
+    editor.on('input', collect);
+    editor.on('beforeinput', beforeInputCollect);
+    editor.setContent(setupHtml);
+    TinySelections.setCursor(editor, setupPath, setupOffset);
+    editor.nodeChanged();
+    TinyContentActions.keydown(editor, Keys.backspace());
+    editor.off('beforeinput', beforeInputCollect);
+    editor.off('input', collect);
+
+    TinyAssertions.assertContent(editor, expectedHtml);
+
+    assert.deepEqual(inputEvents, cancelBeforeInput ? [ 'deleteContentBackward' ] : [ 'deleteContentBackward', 'deleteContentBackward' ]);
+
+    editor.getBody().normalize();
```

Comment:
Why is this here? That seems quite odd so if it's here for a legit reason can you please add a comment?

Reply:
I believe the reason is just that I poached the test from a different file, compressed it, and then forgot to clean out all the things that weren't actually needed. I've removed it.